ABDataCollectionCore.js

```diff
@@ -1106,20 +1113,24 @@ module.exports = class ABDataCollectionCore extends ABMLClass {
                      if (
                         Array.isArray(rowRelateVal) &&
                         rowRelateVal.filter(
-                           (v) => v == values.id || v.id == values.id
+                           (v) =>
+                              v == values.id ||
+                              v.id == values.id ||
+                              v[PK] == values.id
                         ).length > 0 &&
                         !isRelated(updateRelateVal, d.id, PK)
                      ) {
                         updateItemData[f.relationName()] = rowRelateVal.filter(
-                           (v) => (v.id || v) != values.id
+                           (v) => (v.id || v || v[PK]) != values.id
```

Comment:
I think it might be better if we can pull out data from `v[PK]` before `v`.

Should It be `v.id || v[PK] || v` instead ?

Reply:
Good point...thank you.